cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerLoggingConfigurationDiff.java

```diff
@@ -0,0 +1,186 @@
+     * @param desired desired logging configuration, may be null if the related ConfigMap does not exist yet or no changes are required
+     * @param brokerConfigs current configuration
+     * @return Collection of AlterConfigOp containing all entries which were changed from current in desired configuration
+     */
+    private static Collection<AlterConfigOp> diff(int brokerId, String desired,
+                                                  Config brokerConfigs) {
+        if (brokerConfigs == null || desired == null) {
+            return Collections.emptyList();
+        }
+        Map<String, String> currentMap;
+        Collection<AlterConfigOp> updatedCE = new ArrayList<>();
+        currentMap = brokerConfigs.entries().stream().collect(
+            Collectors.toMap(
+                ConfigEntry::name,
+                configEntry -> configEntry.value() == null ? "null" : configEntry.value()));
+
+        OrderedProperties orderedProperties = new OrderedProperties();
+        desired = desired.replaceAll("log4j\\.logger\\.", "");
+        orderedProperties.addStringPairs(desired);
+        Map<String, String> desiredMap = orderedProperties.asMap();
+
+        JsonNode source = patchMapper().configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true).valueToTree(currentMap);
+        JsonNode target = patchMapper().configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true).valueToTree(desiredMap);
+        JsonNode jsonDiff = JsonDiff.asJson(source, target);
+
+        for (JsonNode d : jsonDiff) {
+            String pathValue = d.get("path").asText();
+            String pathValueWithoutSlash = pathValue.substring(1);
+
+            Optional<ConfigEntry> optEntry = brokerConfigs.entries().stream()
+                    .filter(configEntry -> configEntry.name().equals(pathValueWithoutSlash))
+                    .findFirst();
+
+            if (pathValueWithoutSlash.equals("log4j.rootLogger")) {
+                if (!desiredMap.get(pathValueWithoutSlash).matches(".+,.+")) {
+                    log.warn("Broker {} logging: Logger log4j.rootLogger should contain level and appender, e.g. \'log4j.rootLogger = INFO, CONSOLE\'", brokerId);
+                }
+            }
+            String op = d.get("op").asText();
+            if (optEntry.isPresent()) {
+                ConfigEntry entry = optEntry.get();
+                if ("remove".equals(op)) {
+                    removeProperty(updatedCE, pathValueWithoutSlash, entry);
+                } else if ("replace".equals(op)) {
+                    // entry is in the current, desired is updated value
+                    if (!entry.value().equals(parseLogLevelFromAppenderCouple(desiredMap.get(entry.name())))) {
+                        updateOrAdd(entry.name(), desiredMap, updatedCE);
+                    }
+                }
+            } else {
+                if ("add".equals(op)) {
+                    if (pathValueWithoutSlash.equals("log4j.rootLogger")) {
```

Comment:
Any reason why you can't change this in the `desired` map like you do for the ones starting `log4j.logger`?